src/routes/version_creation.rs

```diff
@@ -0,0 +1,239 @@
+        }
+    } else {
+        transaction.commit().await?;
+    }
+
+    result
+}
+
+async fn version_create_inner(
+    mut payload: Multipart,
+    transaction: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+    file_host: &dyn FileHost,
+    uploaded_files: &mut Vec<UploadedFile>,
+) -> Result<HttpResponse, CreateError> {
+    //TODO: Check if mod exists, check if version number is unique
+    let cdn_url = dotenv::var("CDN_URL")?;
+
+    let mut initial_version_data = None;
+    let mut version_builder = None;
+
+    while let Some(item) = payload.next().await {
+        let mut field: Field = item.map_err(CreateError::MultipartError)?;
+        let content_disposition = field.content_disposition().ok_or_else(|| {
+            CreateError::MissingValueError("Missing content disposition".to_string())
+        })?;
+        let name = content_disposition
+            .get_name()
+            .ok_or_else(|| CreateError::MissingValueError("Missing content name".to_string()))?;
+
+        if name == "data" {
+            let mut data = Vec::new();
+            while let Some(chunk) = field.next().await {
+                data.extend_from_slice(&chunk.map_err(CreateError::MultipartError)?);
+            }
+
+            let version_create_data: InitialVersionData = serde_json::from_slice(&data)?;
+            initial_version_data = Some(version_create_data.clone());
+
+            let mod_id: ModId = version_create_data.mod_id.ok_or_else(|| {
+                CreateError::InvalidInput("Mod id is required for version data".to_string())
+            })?;
+
+            let results = sqlx::query!(
+                "SELECT EXISTS(SELECT 1 FROM mods WHERE id=$1)",
+                mod_id.0 as i64
+            )
+            .fetch_one(&mut *transaction)
+            .await?;
+
+            if !results.exists.unwrap_or(true) {
+                return Err(CreateError::InvalidInput(
+                    "An invalid mod id was supplied".to_string(),
+                ));
+            }
+
+            let version_id: VersionId = models::generate_version_id(transaction).await?.into();
+            let body_url = format!("data/{}/changelogs/{}/body.md", mod_id, version_id);
+
+            let uploaded_text = file_host
+                .upload_file(
+                    "text/plain",
+                    &body_url,
+                    version_create_data.version_body.clone().into_bytes(),
+                )
+                .await?;
+
+            uploaded_files.push(UploadedFile {
+                file_id: uploaded_text.file_id.clone(),
+                file_name: uploaded_text.file_name.clone(),
+            });
+
+            // TODO: do a real lookup for the channels
+            let release_channel = match version_create_data.release_channel {
+                VersionType::Release => models::ChannelId(0),
+                VersionType::Beta => models::ChannelId(2),
+                VersionType::Alpha => models::ChannelId(4),
```

Comment:
I made a minor change to the version that this was copied from to fix the sqlx errors.  The release channel ids start at 1, so `Release` should be 1, `Beta` should be 3, and `Alpha` should be 5.